### PR TITLE
Fix Kado Plain bank linking glitchiness

### PR DESCRIPTION
**Pull request by paullinator**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [ ] Yes
- [ ] No

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [ ] Tested on Android device
- [x] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207057109616480

### CODE REVIEW

File: src/plugins/gui/providers/kadoProvider.ts [hunk 1]
```diff
@@ -613,10 +616,32 @@ export const kadoProvider: FiatProviderFactory = {
             }
             console.log('Launching Kado webview url=' + url.href)
 
+            // If Kado needs to launch the Plaid bank linking widget, it needs it in an external
+            // webview to prevent some glitchiness. When needed, Kado will send an onMessage
+            // trigger with the url to open. The below code is derived from Kado's sample code
+            const onMessage = (data: string) => {
+              try {
+                datelog(`**** Kado onMessage ${data}`)
```
Comment (samholmes) [inline src/plugins/gui/providers/kadoProvider.ts:624]: is data for sure not sensitive and therefore okay to log?

File: src/plugins/gui/providers/kadoProvider.ts [hunk 2]
```diff
@@ -613,10 +616,32 @@ export const kadoProvider: FiatProviderFactory = {
...
+                const message = JSON.parse(data)
```
Comment (samholmes) [inline src/plugins/gui/providers/kadoProvider.ts:625]: We should probably be using cleaners for this.

File: src/plugins/gui/providers/kadoProvider.ts [hunk 3]
```diff
@@ -613,10 +616,32 @@ export const kadoProvider: FiatProviderFactory = {
...
+                if (message && message?.type === 'PLAID_NEW_ACH_LINK') {
+                  const achLink = message?.payload?.link
+                  showUi.openExternalWebView({ url: achLink }).catch(e => {
```
Comment (samholmes) [inline src/plugins/gui/providers/kadoProvider.ts:628]: This archLink could be undefined and not string. How did this not error on the static type checker?

**Review by samholmes — CHANGES_REQUESTED** [review]
This may be a "quick fix" but I think we need to be safer when handling the IO here.

File: src/plugins/gui/providers/kadoProvider.ts (hunk 1 above)
Comment (paullinator) [reply src/plugins/gui/providers/kadoProvider.ts:624]: For sure. it's just a url link to redirect the user

File: src/plugins/gui/providers/kadoProvider.ts (hunk 3 above)
Comment (paullinator) [reply src/plugins/gui/providers/kadoProvider.ts]: not sure. but fixed with cleaner

**Review by samholmes — APPROVED** [review]

**Pull request merged**